rusha.sweet.js

```diff
@@ -300,6 +312,69 @@
       return hex(rawDigest(str).buffer);
     };
 
+    this.start = function () {
+      initState(self.heap, self.padMaxChunkLen);
+      return this;
+    };
+
+    this.append = function (chunk) {
+      var chunkOffset = 0;
+      var chunkLen = chunk.byteLength || chunk.length || chunk.size || 0;
+      var turnOffset = self.offset % self.maxChunkLen;
+      
+      self.offset += chunkLen;
+      while (chunkOffset < chunkLen) {
+          var inputLen = Math.min(chunkLen - chunkOffset, self.maxChunkLen - turnOffset);
+          write(chunk, chunkOffset, inputLen, turnOffset);
+          turnOffset += inputLen;
+          chunkOffset += inputLen;
+          if (turnOffset === self.maxChunkLen) {
+              self.core.hash(self.maxChunkLen, self.padMaxChunkLen);
+              turnOffset = 0;
+          }
+      }
+      return this;
+    };
+
+    this.getState = function () {
+      var turnOffset = self.offset % self.maxChunkLen,
+          heap;
+      if(!turnOffset){
```

Comment:
It would be great if you could adhere to the code style here, and insert spaces before `{` and `(`, and after `}`.